Build issues with PGI on cheyenne

This is with the following test cases using ctsm1.0.dev008 which uses cime5.6.10. I also tried them with cime master on this date with cime5.7.0-126-g4f41034.

SMS_D_P1x1_Mmpi-serial.1x1_numaIA.I2000Clm50BgcCropGs.cheyenne_pgi
SMS_D.f10_f10_musgs.I2000Clm50SpGs.cheyenne_pgi.clm-default

On master I get the following error:

```
Starting SETUP for test SMS_D_P1x1_Mmpi-serial.1x1_numaIA.I2000Clm50BgcCropGs.cheyenne_pgi with 1 procs
Finished SETUP for test SMS_D_P1x1_Mmpi-serial.1x1_numaIA.I2000Clm50BgcCropGs.cheyenne_pgi in 1.340754 seconds (FAIL). [COMPLETED 1 of 1]
    Case dir: /gpfs/fs1/work/erik/ctsm/cime/scripts/SMS_D_P1x1_Mmpi-serial.1x1_numaIA.I2000Clm50BgcCropGs.cheyenne_pgi.20180815_142037_gxwo0s
    Errors were:
        ERROR: module command /glade/u/apps/ch/opt/lmod/7.5.3/lmod/lmod/libexec/lmod python load ncarenv/1.2 pgi/17.9 ncarcompilers/0.4.1 netcdf/4.5.0 failed with message:
        Lmod has detected the following error: These module(s) exist but cannot be
        loaded as requested: "netcdf/4.5.0"
           Try: "module spider netcdf/4.5.0" to see how to load the module(s).
```

with cime5.6.10 I got the following where LINPACK isn't included in the build (which was originally reported in ESCOMP/ctsm/#442):

```
../../pgi/mpt/debug/nothreads/mct/noesmf/lib//libclm.a(SoilWaterMovementMod.o): In function `soilwatermovementmod_soilwater_moisture_form_':
SoilWaterMovementMod.F90:(.text+0x1e306): undefined reference to `dgtsv_'
../../pgi/mpt/debug/nothreads/mct/noesmf/lib//libclm.a(UrbBuildTempOleson2015Mod.o): In function `urbbuildtempoleson2015mod_buildingtemperature_':
UrbBuildTempOleson2015Mod.F90:(.text+0xd730): undefined reference to `dgesv_'
```

--- a/config/cesm/machines/config_compilers.xml
+++ b/config/cesm/machines/config_compilers.xml
@@ -1022,6 +1022,7 @@ using a fortran linker.
 <compiler MACH="cheyenne" COMPILER="pgi">
   <SLIBS>
     <append> -llapack -lblas </append>
+    <append MPILIB="mpi-serial"> -ldl </append>
   </SLIBS>
 </compiler>
 

--- a/config/cesm/machines/config_compilers.xml
+++ b/config/cesm/machines/config_compilers.xml
@@ -1019,6 +1019,12 @@ using a fortran linker.
   </SLIBS>
 </compiler>
 
+<compiler MACH="cheyenne" COMPILER="pgi">
+  <SLIBS>
+    <append> -llapack -lblas </append>
+  </SLIBS>
+</compiler>
+
 <compiler MACH="cheyenne">
   <CPPDEFS>
     <!-- these flags enable nano timers -->

--- a/config/cesm/machines/config_machines.xml
+++ b/config/cesm/machines/config_machines.xml
@@ -378,7 +378,10 @@
       <modules compiler="gnu" mpilib="mpi-serial">
 	<command name="load">netcdf/4.6.1</command>
       </modules>
-      <modules compiler="!gnu" mpilib="mpi-serial">
+      <modules compiler="pgi" mpilib="mpi-serial">
+	<command name="load">netcdf/4.4.1.1</command>
+      </modules>
+      <modules compiler="intel" mpilib="mpi-serial">
 	<command name="load">netcdf/4.5.0</command>
       </modules>
     </module_system>

